src/bin/rustup-init.rs

```diff
@@ -62,18 +60,7 @@ fn run_rustup_inner() -> Result<utils::ExitCode> {
     utils::current_dir()?;
     utils::current_exe()?;
 
-    // The name of arg0 determines how the program is going to behave
-    let arg0 = match process().var("RUSTUP_FORCE_ARG0") {
-        Ok(v) => Some(v),
-        Err(_) => process().args().next(),
-    }
-    .map(PathBuf::from);
-    let name = arg0
-        .as_ref()
-        .and_then(|a| a.file_stem())
-        .and_then(std::ffi::OsStr::to_str);
-
-    match name {
+    match utils::current_name().as_deref() {
```

Comment:
I think `currentprocess` would be a better place to put this.

Reply:
Is that initialized by the time we get here? I admit I haven't looked at that code, but it looks like it might not be?

Reply:
yes - if its not set anything accessing OS global state will blow up. https://github.com/rust-lang/rustup/blob/master/CONTRIBUTING.md#no-direct-use-of-process-state-outside-rustupcurrentprocess This is how we test even these OS global-state using entry points without having to invoke a full binary every time.